packages/cli/src/commands/mock.ts

```diff
@@ -1,43 +1,97 @@
-    const signaleInteractiveInstance = new signale.Signale({ interactive: true });
-
     const {
-      flags: { port, dynamic, host },
+      flags: { port, dynamic, host, multiprocess },
       args: { spec },
     } = this.parse(Server);
 
-    signaleInteractiveInstance.await('Starting Prism…');
-
-    if (dynamic) {
-      signale.star('Dynamic example generation enabled.');
-    }
+    if (cluster.isMaster) {
+      cluster.setupMaster({ silent: true });
 
-    const server = createServer(spec, { mock: { dynamic } });
-    try {
-      const address = await server.listen(port, host);
+      const signaleInteractiveInstance = new signale.Signale({ interactive: true });
+      signaleInteractiveInstance.await({ prefix: chalk.bgWhiteBright.black('[CLI]'), message: 'Starting Prism…' });
 
-      if (server.prism.resources.length === 0) {
-        signaleInteractiveInstance.fatal('No operations found in the current file.');
-        this.exit(1);
+      if (dynamic) {
+        logCLIMessage(`Dynamic example generation ${chalk.green('enabled')}.`);
       }
 
-      signaleInteractiveInstance.success(`Prism is listening on ${address}`);
+      if (multiprocess) {
+        logCLIMessage(`Multi process ${chalk.green('enabled')}.`);
+
+        const worker = cluster.fork();
 
-      server.prism.resources.forEach(resource => {
-        signale.note(`${resource.method.toUpperCase().padEnd(10)} ${address}${resource.path}`);
-      });
-    } catch (e) {
-      signaleInteractiveInstance.fatal(e.message);
+        if (worker.process.stdout) {
+          pipeOutputToSignale(worker.process.stdout);
+        }
+      } else {
+        const logStream = new PassThrough();
+        const logInstance = createLogger('CLI', undefined, logStream);
+        pipeOutputToSignale(logStream);
+        await createPrismServerWithLogger(spec, dynamic, port, host, logInstance);
+      }
+    } else if (multiprocess && cluster.isWorker) {
+      const logInstance = createLogger('CLI');
+      await createPrismServerWithLogger(spec, dynamic, port, host, logInstance);
     }
   }
 }
+
+async function createPrismServerWithLogger(
+  spec: any,
+  dynamic: boolean,
+  port: number,
+  host: string | undefined,
+  logInstance: Logger,
+) {
+  const server = createServer(spec, { mock: { dynamic } }, logInstance.child({ name: 'HTTP SERVER' }));
+  try {
+    const address = await server.listen(port, host);
+    if (server.prism.resources.length === 0) {
+      logInstance.fatal('No operations found in the current file.');
+      cluster.worker.kill();
+    }
+    server.prism.resources.forEach(resource => {
+      logInstance.note(`${resource.method.toUpperCase().padEnd(10)} ${address}${resource.path}`);
+    });
+    logInstance.start(`Prism is listening on ${address}`);
+  } catch (e) {
+    logInstance.fatal(e.message);
+    cluster.worker.kill();
+  }
+}
+
+function pipeOutputToSignale(stream: Readable) {
+  function constructPrefix(logLine: LogDescriptor): string {
```

Comment:
hm.. if I'm not mistaken, we can just move out `constructPrefix` from `pipeOutputToSignale` (?)

Reply:
Yeah, I'll probably refactor this command and offload most of the shit in the `createServer` file and go from there, hopefully that's going to help a lot in this command specifically. Stay tuned for updates.

Reply:
:+1:

Reply:
@XVincentX, I've seen that `pipeOutputToSignale` is now in another file. But can we also `constructPrefix` move out of `pipeOutputToSignale`? I don't see a need for `constructPrefix` to be defined inside `pipeOutputToSignale` (but maybe there is (?)). And also, how about `pipeOutputToSignale` -> `pipeOutputToSignal` ?